### ISSUE TITLE
lolminer not working with beamhash II

### ISSUE THREAD

**Issue by jch9678** [RainbowMiner/RainbowMiner#638]
it looks like it tries to connect to the wrong nicehash pool.  Not sure if it's 1 or 2 issues.
[debug_2019-08-19.zip](https://github.com/RainbowMiner/RainbowMiner/files/3517890/debug_2019-08-19.zip)

**Comment by jch9678** [comment]
I'm pretty sure the pool configuration is the problem.  It's connecting to the wrong nicehash pool so the miner crashes for some reason and then when it tries to restart my machine freezes.  I think my 1-card test machine recovers from the crash but multi-card machines cannot.

**Comment by jch9678** [comment]
The fix implemented in 4.4 is not working.  lolminer is now trying to use this pool EquihashR25x5.usa.nicehash.com

The stratum generator on NiceHash says I should be using this 
stratum+tcp://beamv2.usa-new.nicehash.com:3378

Here are the pools that are available https://www.nicehash.com/algorithm unless there are others not published.

**Comment by jch9678** [comment]
4.4.01 is still not working.  It is now connecting to beam.usa.nicehash.com:3370 not beamv2

I don't know if this makes a difference but I haven't switched to the new platform.  Does RM need any additional input if/when I do switch?  If beamv2 is only offered on the new platform does that matter?

**Comment by jch9678** [comment]
There's also this: WARNING: Pool API (CryptoKnight) for TUBE has failed. 
I think you can use the same debug zip in this Issue.

**Comment by RainbowMiner** [comment]
> I don't know if this makes a difference but I haven't switched to the new platform. Does RM need any additional input if/when I do switch? If beamv2 is only offered on the new platform does that matter?


Yes, that explains the behaviour. There is no BeamV2 on the old platform. Only Beam (for GRIMM). If you want to do the switch:
- (re-)register at nicehash.com. Make sure you use the same email address that you had used on old.nicehash.com, then they will transfer all you funds, when NH finally switches off the old platform.
- now find your new mining bitcoin address
- from here you have two ways to go:

A) **if you don't want to mine on the old platform, anymore**
  - edit pools.config.txt, find "Nicehash" and change your "Wallet" to match the new mining address and set "Platform" to "v2" (optionally set API_Key, API_Secret and OrganizationID).
```
    "BTC": "yournewnhwalletaddress",
    "Platform": "v2",
    "API_Key": "optionalapikey",
    "API_Secret": "optionalapisecret",
    "OrganizationID": "optionalorganizationid"
```

B) **if you want to mine on the old, and the new platform** (RainbowMiner decides about profitability)
  - enable "NiceHashV2" as new pool in config.txt (add to field "PoolName")
  - edit pools.config.txt and fill in all data to "NiceHashV2" section:
```
    "BTC": "yournewnhwalletaddress",
    "API_Key": "optionalapikey",
    "API_Secret": "optionalapisecret",
    "OrganizationID": "optionalorganizationid"
```

**Comment by RainbowMiner** [comment]
I have tried old BEAM on both NiceHash platforms and it turns out, that it sometimes finds shares and sometimes says "over target". It looks like buyers on NH switch wildly between GRIMM and the old BEAM fork (for whatever reason). Since it behaves quiet unstable (and isn't paying good, btw), I have decided to disable old BEAM on both Nicehash platforms. 
Only BEAMv2 will be available, and only on the new platform.
